x-pack/plugins/lens/public/app_plugin/app.tsx

```diff
@@ -403,8 +386,7 @@ export function App({
     attributeService,
     redirectTo,
     chrome.recentlyAccessed,
-    state.persistedDoc?.savedObjectId,
-    state.persistedDoc?.state,
+    state.persistedDoc,
```

Comment:
No strong feelings as I expect this hook to be removed soon anyway, but I think the `state.persistedDoc` dependency here is not necessary and on the contrary causes this initialization logic to run twice. Removing it (with a linter comment) makes the hack of line 360 unnecessary.